### ISSUE TITLE
Shelley: protocol testing with lowering `d`

### ISSUE THREAD

**Issue by edsko** [input-output-hk/ouroboros-network#1770]
@mrBliss 's current integration work and tests a fixed value for `d` (#1695). The Shelley testnet will want to start reducing `d` fairly quickly (after all, a major reason for the testnet is to test staking). It is therefore important that we test soon that we can lower this parameter. Doing so is however non-trivial as the ledger rules impose restrictions on precisely when this can happen, and so the test infrastructure will need to make sure that these restrictions are satisfied before submitting updates to change `d`.

Somewhat ironically, this is a lower priority for the mainnet, as there it will probably take longer before we start reducing `d`.

**Comment by nfrisby** [comment]
Edsko shared on Slack:

> comment from a meeting just now that is relevant to you: "overlay schedule should be considered a "fallback" in a way for the people who have not yet delegated, so the value of d should not be set too low when not enough stake has been delegated yet" (this is something that the people who vary d will ensure, and so the tests should try to do something similar)
